File: src/leetcode/m240/SearchA2DMatrixII.go
package m240

// LeetCode 240
func searchMatrix(matrix [][]int, target int) bool {
	// Start from the top-right corner
	row, col := 0, len(matrix[0])-1

	for row < len(matrix) && col >= 0 {
		if matrix[row][col] == target {
			return true
		} else if matrix[row][col] > target {
			col-- // Move left
		} else {
			row++ // Move down
		}
	}

	return false
}

File: src/leetcode/m240/SearchA2DMatrixII.java
/*
 * Copyright (c) 2024 THGEEK.COM. All rights reserved.
 */

package leetcode.m240;

/**
 * LeetCode 240
 *
 * @author Tao Hong
 * @version 1.0
 * @since 2024/11/13 09:32
 */
public class SearchA2DMatrixII {
    public boolean searchMatrix(int[][] matrix, int target) {
        int i = 0;
        int j = matrix[0].length - 1;
        while (i < matrix.length && j >= 0) {
            if (matrix[i][j] == target) {
                return true;
            } else if (matrix[i][j] > target) {
                j--;
            } else {
                i++;
            }
        }
        return false;
    }
}
